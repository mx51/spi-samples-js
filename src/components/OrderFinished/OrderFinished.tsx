import React, { useEffect, useState } from 'react';
import { Grid } from '@material-ui/core';
import { useSelector } from 'react-redux';
import selectedTerminalIdSelector from '../../redux/reducers/SelectedTerminalSlice/selectedTerminalSliceSelector';
import { ITerminalProps } from '../../redux/reducers/TerminalSlice/interfaces';
import {
  terminalInstance,
  terminalTransactionTypeObject,
} from '../../redux/reducers/TerminalSlice/terminalsSliceSelectors';
import FlowPanel from '../FlowPanel';
import Layout from '../Layout';
import PurchaseFlow from '../PurchaseFlow';
import { PaymentSummary } from '../PaymentSummary/PaymentSummary';
import { TxLogItem, TxLogService } from '../../services/txLogService';

function OrderFinished(): React.ReactElement {
  const selectedTerminal = useSelector(selectedTerminalIdSelector);
  const currentTerminal = useSelector(terminalInstance(selectedTerminal)) as ITerminalProps;
  const { typePath } = useSelector(terminalTransactionTypeObject(selectedTerminal));
  const [currentTransaction, setcurrentTransaction] = useState<TxLogItem>();

  const currentTxId = currentTerminal?.txFlow?.posRefId;

  useEffect(() => {
    if (!currentTransaction && currentTxId) {
      setcurrentTransaction(TxLogService.findTxByPosRefId(currentTxId));
    }
  }, [currentTransaction]);

  if (!currentTransaction) return <div>Could not load transaction.</div>;

  return (
    <Layout>
      <PaymentSummary currentTransaction={currentTransaction} typePath={typePath} />
      <Grid item xs={4}>
        <FlowPanel>
          <PurchaseFlow />
        </FlowPanel>
      </Grid>
    </Layout>
  );
}

export default OrderFinished;
