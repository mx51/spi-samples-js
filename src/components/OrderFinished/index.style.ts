import { createStyles, makeStyles, Theme } from '@material-ui/core';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    receiptBoxWrapper: {
      height: '100%',
      maxHeight: '100%',
      display: 'flex',
      alignItems: 'center',
      padding: theme.spacing(6, 0),
      '& > div': {
        background: theme.palette.common.white,
        border: `1px solid ${theme.palette.secondary.main}`,
        borderRadius: theme.spacing(0.5),
        width: '100%',
        height: '100%',
        overflowY: 'scroll', // handle flow status longer messages
        padding: theme.spacing(2),
        '& > h1': {
          paddingBottom: theme.spacing(2),
        },
      },
    },
  })
);
export default useStyles;
