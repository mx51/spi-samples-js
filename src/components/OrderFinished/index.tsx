import React from 'react';
import { Grid } from '@material-ui/core';

import FlowPanel from '../FlowPanel';
import Layout from '../Layout';
import PaymentSummary from '../PaymentSummary';
import PurchaseFlow from '../PurchaseFlow';
import ReceiptPanel from '../TerminalsPage/TerminalDetails/ReceiptPanel';
import useStyles from './index.style';

function OrderFinished(): React.ReactElement {
  const classes = useStyles();

  return (
    <Layout>
      <Grid container spacing={2}>
        <Grid item xs={5}>
          <PaymentSummary />
        </Grid>
        <Grid item xs={3}>
          <ReceiptPanel />
        </Grid>
        <Grid item xs={4}>
          <FlowPanel>
            <PurchaseFlow />
          </FlowPanel>
          {/* <Order disablePayNow /> */}
        </Grid>
      </Grid>
    </Layout>
  );
}

export default OrderFinished;
