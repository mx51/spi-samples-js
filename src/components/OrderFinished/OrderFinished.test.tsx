import React from 'react';
import * as redux from 'react-redux';
import OrderFinished from '.';

import mockWithRedux from '../../utils/tests/common';

describe('Test <OrderFinished />', () => {
  test('snapshot for OrderFinishedPage component', () => {
    const spySelector = jest.spyOn(redux, 'useSelector');
    spySelector.mockReturnValue({
      posId: '123',
      txFlow: {
        finished: true,
        success: 'Success',
        request: {
          data: {
            purchaseAmount: 550,
            surchargeAmount: 100,
            cashAmount: 200,
            tipAmount: 150,
          },
        },
        response: {
          data: {
            merchantReceipt: 'jfkgjf',
          },
        },
      },
    });
    const { container } = mockWithRedux(<OrderFinished />);
    expect(container).toMatchSnapshot();
  });
});
