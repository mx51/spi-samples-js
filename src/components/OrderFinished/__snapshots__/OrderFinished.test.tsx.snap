// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <OrderFinished /> snapshot for OrderFinishedPage component 1`] = `undefined`;
